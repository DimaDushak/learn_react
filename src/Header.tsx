import * as React from 'react';

function Header(): JSX.Element {
    return <div>Hello world</div>;
}

export default Header;
