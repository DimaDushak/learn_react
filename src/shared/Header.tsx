import React from 'react';
import styles from './header.css';

function Header(): JSX.Element {
    return <div className={styles.example}>Hello world</div>;
}

export default Header;
