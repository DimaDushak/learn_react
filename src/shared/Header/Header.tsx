import React from 'react';
import styles from './header.css';
import SearchBlock from './SearchBlock';
import SortBlock from './SortBlock';
import TitleBlock from './TitleBlock';

export default function Header(): JSX.Element {
    return (
        <header className={styles.header}>
            <SearchBlock />
            <TitleBlock />
            <SortBlock />
        </header>
    );
}
