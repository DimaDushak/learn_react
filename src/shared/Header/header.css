@media (min-width: 1024px) {
    .header {
        display: flex;
        justify-content: space-between;
    }
}
