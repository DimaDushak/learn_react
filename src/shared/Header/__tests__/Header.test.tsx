import { render } from '@testing-library/react';
import React from 'react';
import Header from '../Header';

test('should render Header (snapshot)', () => {
    const { container } = render(<Header />);

    expect(container.firstChild).toMatchSnapshot();
});
