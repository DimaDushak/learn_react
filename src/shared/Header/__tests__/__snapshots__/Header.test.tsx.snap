// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header (snapshot) 1`] = `
<header
  class="header"
>
  <div
    class="searchBlock"
  >
    <a
      class="userBlock userBlock"
      href="https://www.reddit.com/api/v1/authorize?client_id=undefined&response_type=code&state=random_string&redirect_uri=http://localhost:3000/auth&duration=permanent&scope=read identity submit"
      rel="noreferrer"
    >
      <img
        alt="User avatar"
        class="authorAvatar"
        src="/static/image/AnonimIcon.svg"
      />
      <span
        class="anonim"
      >
        Аноним
      </span>
    </a>
  </div>
  <div
    class="titleBlock"
  >
    title
  </div>
  <div>
    sort
  </div>
</header>
`;
